docker/android-system.sh

```diff
@@ -46,64 +79,474 @@ EOF
+    sync external/mksh
+    sync external/pcre
+    sync external/safe-iop
+    sync external/zlib
+    sync libnativehelper
+    sync prebuilts/clang/linux-x86/host/3.6
+    sync prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.15-4.8
+    sync prebuilts/misc
+    sync system/core
+
+    case "${ARCH}" in
+        arm)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+        ;;
+        arm64)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
+
+    # avoid build tests
+    rm bionic/linker/tests/Android.mk
+    rm bionic/tests/Android.mk
+    rm bionic/benchmarks/Android.mk
+    # we don't need the relocation packer, and removing
+    # the unittests from it is a bit of work.
+    rm bionic/tools/relocation_packer/Android.mk
+}
+
+# tested on 7.0.0_r36 (SDK 24)
+# tested on 7.1.2_r39 (SDK 25, not supported)
+#   API level 25, requires for Android 7.1, is not provided in NDKs
+android_repo_v7() {
+    sync bionic
+    sync build
+    sync build/kati
+    sync external/boringssl
+    sync external/compiler-rt
+    sync external/elfutils
+    sync external/jemalloc
+    sync external/libcxx
+    sync external/libcxxabi
+    sync external/libselinux
+    sync external/libunwind
+    sync external/libunwind_llvm
+    sync external/llvm
+    sync external/mksh
+    sync external/pcre
+    sync external/safe-iop
+    sync external/zlib
+    sync prebuilts/clang/host/linux-x86
+    sync prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.15-4.8
+    sync prebuilts/misc
+    sync prebuilts/ndk
+    sync prebuilts/ninja/linux-x86
+    sync system/core
+
+    case "${ARCH}" in
+        arm)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+        ;;
+        arm64)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
+
+    # avoid build tests
+    rm bionic/linker/tests/Android.mk
+    rm bionic/tests/Android.mk
+    rm bionic/benchmarks/Android.mk
+    rm prebuilts/misc/common/android-support-test/Android.mk
+    # we don't need the relocation packer, and removing
+    # the unittests from it is a bit of work.
+    rm bionic/tools/relocation_packer/Android.mk
+
+    remove_tests
+}
+
+# tested on 8.0.0_r51 (SDK 26)
+# tested on 8.1.0_r81 (SDK 27)
+android_repo_v8() {
+    # need to build LLVM components, or libLLVM is disabled.
+    export FORCE_BUILD_LLVM_COMPONENTS=true
```

Comment:
Android 8 is a pain because it now requires the LLVM components, but by default turns off building them from source, and using system packages doesn't work. We therefore need to force their compilation, making these builds quite sluggish but overall, work fine.